File: javascript/sample4.test.js
import Plateau from "./plateau";
import MarsRover from "./mars-rover";

describe('Mars Rover', function() {

    it('can follow instructions', function () {
        var plateau = new Plateau(5, 5);
        var rover = new MarsRover(plateau, [1, 2], "N");

        rover.turnLeft();
        expect(rover.heading).toEqual("W");
        rover.turnLeft();
        expect(rover.heading).toEqual("S");
        rover.turnLeft();
        expect(rover.heading).toEqual("E");
        rover.forward();
        expect(rover.position).toEqual([2, 2]);
        rover.backward();
        expect(rover.position).toEqual([1, 2]);
        rover.turnRight();
        expect(rover.position).toEqual([1, 2]);
        expect(rover.heading).toEqual("S");

    });

});

File: go/marsroverkata/sample4_test.go
package marsrover

import (
	"github.com/stretchr/testify/assert"
	"testing"
)

func TestCanMove(t *testing.T) {
	plateau := Plateau{maxX: 5, maxY: 5}
	startingPosition := Coordinates{1,2}
	marsRover := MarsRover{plateau: plateau, heading: N, position: startingPosition}

	marsRover.turnLeft()
	assert.Equal(t, W, marsRover.heading)
	marsRover.turnLeft()
	assert.Equal(t, S, marsRover.heading)
	marsRover.turnLeft()
	assert.Equal(t, E, marsRover.heading)
	marsRover.forward()
	assert.Equal(t, "2 2 E", marsRover.currentLocation())
	marsRover.backward()
	assert.Equal(t, "1 2 E", marsRover.currentLocation())
	marsRover.turnRight()
	assert.Equal(t, "1 2 S", marsRover.currentLocation())
}
